specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json

```diff
@@ -0,0 +1,870 @@
+        "imageFamily": {
+          "type": "string",
+          "description": "ImageFamily for categorization in classic portal."
+        },
+        "iconUri": {
+          "type": "string",
+          "description": "IconUri of the image, must be a valid URI."
+        },
+        "smallIconUri": {
+          "type": "string",
+          "description": "Small thumbnail IconUri of the image, must be a valid URI."
+        },
+      },
+      "required": [
+        "imageType"
+      ],
+      "description": "Shared VM image classic compute profile."
+    },
+    "SharedVMImagePurchasePlan": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Purchase plan name."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "Publisher name."
+        },
+        "product": {
+          "type": "string",
+          "description": "Product name."
+        }
+      },
+      "required": [
+      ],
+      "description": "Shared VM image purchase plan."
+    },
+	"SharedVMImageVersion": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SharedVMImageVersionProperties"
+        }
+      },
+      "allOf": [{
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Shared VM image verison resource."
+    },
+	 "SharedVMImageVersionProperties": {
+      "properties": {
+        "classicComputeProfile": {
+          "$ref": "#/definitions/SharedVMImageVersionClassicComputeProfile",
+          "description": "Properties of the image version classic compute profile."
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/SharedVMImageVersionStorageProfile",
+          "description": "Properties of the image version storage profile."
+        },
+		"publishingProfile": {
+          "$ref": "#/definitions/SharedVMImageVersionPublishingProfile",
+          "description": "Properties of the image version publishing profile."
+        },
+		"replicationStatus": {
+          "$ref": "#/definitions/ReplicationStatus",
+          "description": "Properties of the image version replication status."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string"
+        },
+      },
+      "required": [
+        "publishingProfile",
+		"storageProfile",
+		"classicComputeProfile"
+      ],
+      "description": "Shared VM image version resource properties."
+    },
+	"SharedVMImageVersionClassicComputeProfile": {
+      "properties": {
+        "showInGui": {
+          "type": "boolean",
+          "description": "Used by portal to show the image in dropdown."
+        },
+        "imageName": {
+          "type": "string",
+          "description": "ImageName for classic compute. If not specified we publish the image with Guid name and hide the image in listing in classic compute."
+        }
+      },
+      "required": [
```

Comment:
empty `required` not valid according to spec, please remove (or make something required 😉 )